test/integration/kubernetes_deploy_test.rb

```diff
@@ -806,6 +806,46 @@ def test_invalid_context
     end
   end
 
+  def test_multiple_configuration_files
+    old_config = ENV['KUBECONFIG']
+    begin
+      config_file = File.join(__dir__, '../fixtures/kube-config/unknown_config.yml')
+      ENV['KUBECONFIG'] = config_file
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
+        'Result: FAILURE',
+        'Configuration invalid',
+        "Kube config not found at #{config_file}"
+      ], in_order: true)
+
+      ENV['KUBECONFIG'] = " : "
+      result = deploy_fixtures('hello-cloud')
+      assert_deploy_failure(result)
+      assert_logs_match_all([
```

Comment:
We should probably reset the logger between these subtests. Otherwise, technically the first two assertions here could pass based on the output of the previous subtest. Basically you need to do `@logger_stream = StringIO.new`. Could be nice to make a `reset_logger` helper for this in `test_helper.rb`.

Reply:
Helper function reset_logger is added to clean StringIO buffer.